server/src/main/java/org/opensearch/action/DocWriteRequest.java

```diff
@@ -248,6 +249,25 @@ static DocWriteRequest<?> readDocumentRequest(@Nullable ShardId shardId, StreamI
         return docWriteRequest;
     }
 
+    /**
+     * Validates whether the doc id length is under the limit.
+     * @param id DocId to verify
+     * @param validationException containing all the validation errors.
+     * @return validationException
+     */
+    static ActionRequestValidationException validateDocIdLength(String id, ActionRequestValidationException validationException) {
+        if (id != null && UnicodeUtil.calcUTF16toUTF8Length(id, 0, id.length()) > 512) {
+            return addValidationError(
+                "id ["
+                    + id
+                    + "] is too long, must be no longer than 512 bytes but was: "
+                    + UnicodeUtil.calcUTF16toUTF8Length(id, 0, id.length()),
```

Comment:
I don't think we need to calculate length twice (`if` and here), could we store it in local variable?

Reply:
Done.